src/goto-instrument/cover.cpp

```diff
@@ -1248,23 +1238,22 @@ void instrument_cover_goals(
           i_it->make_assertion(d);
           i_it->source_location=source_location;
           i_it->source_location.set_comment(comment_t);
-          i_it->source_location.set(ID_coverage_criterion, coverage_criterion);
-          i_it->source_location.set_property_class(property_class);
+          i_it->source_location.set_property_class("coverage");
 
           const std::string comment_f="decision `"+d_string+"' false";
           goto_program.insert_before_swap(i_it);
           i_it->make_assertion(not_exprt(d));
           i_it->source_location=source_location;
           i_it->source_location.set_comment(comment_f);
-          i_it->source_location.set(ID_coverage_criterion, coverage_criterion);
-          i_it->source_location.set_property_class(property_class);
+          i_it->source_location.set_property_class("coverage");
         }
         
         for(std::size_t i=0; i<decisions.size()*2; i++)
           i_it++;
       }
       break;
-      
+
+    case coverage_criteriont::BOUNDARY:
     case coverage_criteriont::MCDC:
```

Comment:
Sent the implementation to Lucas. While I only changed several lines of codes, the diff seems to be quite big. 

The 'cover.cpp' becomes more complex day by day. Thinking to refactor the coverage codes. A class like 'class coveraget'?